File: tests/fixtures/edgecase_repo/kotlin/Service5.kt
package com.example.gen

import com.example.gen.Base5

interface Handler5 {
    fun handle(payload: String): Boolean
}

open class Base5(val name: String)

class Service5(name: String) : Base5(name), Handler5 {
    override fun handle(payload: String): Boolean {
        validate(payload)
        return true
    }

    private fun validate(payload: String) {}
}

fun helper5(value: String): String {
    return value
}

File: tests/fixtures/edgecase_repo/csharp/Service5.cs
using System;

namespace Example.Gen
{
    public interface IHandler5
    {
        bool Handle(string payload);
    }

    public class Base5
    {
        protected string Name;
    }

    public class Service5 : Base5, IHandler5
    {
        public Service5(string name)
        {
            this.Name = name;
        }

        public bool Handle(string payload)
        {
            this.Validate(payload);
            return true;
        }

        private void Validate(string payload) { }
    }
}
